vms/platformvm/state/state_test.go

```diff
@@ -988,43 +753,49 @@ func TestValidatorWeightDiff(t *testing.T) {
 	}
 }
 
-// Tests PutCurrentValidator, DeleteCurrentValidator, GetCurrentValidator,
-// ApplyValidatorWeightDiffs, ApplyValidatorPublicKeyDiffs
-func TestStateAddRemoveValidator(t *testing.T) {
+func TestState_ApplyValidatorDiffs(t *testing.T) {
 	require := require.New(t)
 
 	state := newTestState(t, memdb.New())
 
 	var (
-		numNodes  = 3
-		subnetID  = ids.GenerateTestID()
-		startTime = time.Now()
-		endTime   = startTime.Add(24 * time.Hour)
-		stakers   = make([]Staker, numNodes)
+		numNodes       = 5
+		subnetID       = ids.GenerateTestID()
+		startTime      = time.Now()
+		endTime        = startTime.Add(24 * time.Hour)
+		primaryStakers = make([]Staker, numNodes)
+		subnetStakers  = make([]Staker, numNodes)
 	)
-	for i := 0; i < numNodes; i++ {
-		stakers[i] = Staker{
+	for i := range primaryStakers {
+		sk, err := bls.NewSecretKey()
+		require.NoError(err)
+
+		primaryStakers[i] = Staker{
 			TxID:            ids.GenerateTestID(),
 			NodeID:          ids.GenerateTestNodeID(),
+			PublicKey:       bls.PublicFromSecretKey(sk),
+			SubnetID:        constants.PrimaryNetworkID,
 			Weight:          uint64(i + 1),
 			StartTime:       startTime.Add(time.Duration(i) * time.Second),
 			EndTime:         endTime.Add(time.Duration(i) * time.Second),
 			PotentialReward: uint64(i + 1),
 		}
-		if i%2 == 0 {
-			stakers[i].SubnetID = subnetID
-		} else {
-			sk, err := bls.NewSecretKey()
-			require.NoError(err)
-			stakers[i].PublicKey = bls.PublicFromSecretKey(sk)
-			stakers[i].SubnetID = constants.PrimaryNetworkID
+	}
+	for i, primaryStaker := range primaryStakers {
+		subnetStakers[i] = Staker{
+			TxID:            ids.GenerateTestID(),
+			NodeID:          primaryStaker.NodeID,
+			PublicKey:       nil, // Key is inherited from the primary network
+			SubnetID:        subnetID,
+			Weight:          uint64(i + 1),
+			StartTime:       primaryStaker.StartTime,
+			EndTime:         primaryStaker.EndTime,
+			PotentialReward: uint64(i + 1),
```

Comment:
doesn't seem like we use `PotentialReward` anyway, but if start time and end time are the same, then shouldn't the potential reward be 0?

Reply:
> if start time and end time are the same

`endTime` is 24h after `startTime` throughout this test, but you're right these values don't really matter for this test.